### PR TITLE
doc: suggestions for OSSL_PROVIDER_load_ex design document

### PR DESCRIPTION
**Pull request by mspncp**
**doc: suggestions for OSSL_PROVIDER_load_ex design document**
    
    Late review comments for pull request #21604, sort of.

**doc: change name of OSSL_PROVIDER_load_ex design document**
    
    Use dashes instead of underscores, to be more consistent with
    existing document names. And speaking of consistency, introduce
    a consistent name transformation, which will scale better when
    design documents start filling the folder ;-)
    
        OSSL_PROVIDER_load_ex -> ossl-provider-load-ex.md

### CODE REVIEW

**Comment by beldmit** [comment]
A doc-nit failure is relevant :(

File: doc/designs/ossl-provider-load-ex.md
Comment (beldmit) [inline doc/designs/ossl-provider-load-ex.md:2]: Not sure. The document originally was intended to describe all the possible options, not only the new one.

File: doc/designs/ossl-provider-load-ex.md
Comment (beldmit) [inline doc/designs/ossl-provider-load-ex.md:50]: "simply returns" instead of future tense?

File: doc/designs/ossl-provider-load-ex.md
Comment (beldmit) [inline doc/designs/ossl-provider-load-ex.md:53]: Probably it's worth mentioning that the parameters from the configuration file will be completely ignored.

**Comment by beldmit** [comment]
@mspncp you provided much better wording than I did, and I mostly agree with your amendments

File: doc/designs/ossl-provider-load-ex.md
Comment (mspncp) [reply doc/designs/ossl-provider-load-ex.md:53]: Oh, then I misunderstood. I thought the extra parameters would be merged with ones from the config file, overwriting existing value.

Comment (beldmit) [reply doc/designs/ossl-provider-load-ex.md:53]: No. All-or-nothing.

File: doc/designs/ossl-provider-load-ex.md
Comment (mspncp) [reply doc/designs/ossl-provider-load-ex.md:2]: I don't quite follow. This here is just a design document, not the official documentation or a manual page. You proposed a new API, `OSSL_PROVIDER_load_ex` and this is the design document for it. Mentioning the alternatives serves as a rationale for your design decisions but it is not the goal of this document to document all existing options. I would expect those to be already documented in detail in our manpages ;-)

Comment (beldmit) [reply doc/designs/ossl-provider-load-ex.md:2]: Fair point, I agree, thanks!

**Comment by mspncp** [comment]
I did a rebase [without tree-changes](https://github.com/openssl/openssl/compare/c4f060afbbc784f03cb0d737a85ce5b4210dcfd3..40f4c757cb45393240102a3a5e30d77c6eaa8526) to autosquash my fixups.  This was my [last fixup](https://github.com/openssl/openssl/commit/c4f060afbbc784f03cb0d737a85ce5b4210dcfd3).

**Review by beldmit — APPROVED** [review]
LGTM if tests passed

**Comment by beldmit** [comment]
Ping @openssl/committers for the 2nd review

**Review by t8m — APPROVED** [review]

File: doc/designs/ossl-provider-load-ex.md
Comment (DDvO) [inline doc/designs/ossl-provider-load-ex.md:43]: A question: so far, the doc said that at least scenarios 1. and 2. are possible.
This PR suggests that these are the only ones. Is this correct?

Comment (beldmit) [reply doc/designs/ossl-provider-load-ex.md:43]: Yes. The combination of the config and app-defined parameters is not supported.

**Comment by beldmit** [comment]
Merged. Many thanks for the clarification of the document!

**Pull request closed without merging**
